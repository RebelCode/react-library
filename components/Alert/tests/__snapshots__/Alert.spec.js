import React from "react";
import { render, cleanup } from "@testing-library/react";
import { Alert } from "../../index";

afterEach(cleanup);

describe("Alert", () => {
  it("should take a snapshot", () => {
    const { asFragment } = render(
      <Alert aria-label="Alert" type={"success"} />
    );

    expect(asFragment(<Alert type={"success"} />)).toMatchSnapshot();
  });

  it("renders correctly", () => {
    const { queryByLabelText } = render(<Alert aria-label="Alert" />);

    expect(queryByLabelText("Alert")).toBeTruthy();
  });

  it("renders text correctly", () => {
    const { queryByText } = render(
      <Alert aria-label="Alert" children="Success" />
    );

    expect(queryByText("Success")).toBeTruthy();
    expect(queryByText("Failed")).toBeFalsy();
  });
});
