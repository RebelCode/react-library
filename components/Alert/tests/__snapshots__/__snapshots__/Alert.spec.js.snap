// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Alert should take a snapshot 1`] = `
<DocumentFragment>
  <div
    aria-label="Alert"
    class="success"
  >
    <div
      class="content"
    />
  </div>
</DocumentFragment>
`;
